metadata-integration/java/openlineage-converter/src/main/java/io/datahubproject/openlineage/converter/OpenLineageToDataHub.java

```diff
@@ -632,13 +632,130 @@ private static void convertJobToDataJob(
+            tableName = name;
+            log.info("Using table name directly from output dataset name: {}", tableName);
+            break;
+          }
+        }
+      }
+      
+      // Method 3: Check for table identifiers in symlinks
+      if (tableName == null && event.getOutputs() != null) {
+        for (OpenLineage.OutputDataset output : event.getOutputs()) {
+          if (output.getFacets() != null && output.getFacets().getSymlinks() != null) {
+            for (OpenLineage.SymlinksDatasetFacetIdentifiers symlink : 
+                 output.getFacets().getSymlinks().getIdentifiers()) {
+              if ("TABLE".equals(symlink.getType())) {
+                String name = symlink.getName();
+                if (name != null) {
+                  // Handle table/name format
+                  if (name.startsWith("table/")) {
+                    name = name.replaceFirst("table/", "").replace("/", ".");
+                  }
+                  tableName = name;
+                  log.info("Extracted table name from symlink: {}", tableName);
+                  break;
+                }
+              }
+            }
+            if (tableName != null) break;
+          }
+        }
+      }
+      
+      // Method 4: Extract table name from warehouse paths (as a last resort)
+      if (tableName == null && event.getOutputs() != null) {
+        for (OpenLineage.OutputDataset output : event.getOutputs()) {
+          String path = output.getName();
+          if (path != null && path.contains("/warehouse/")) {
+            // Extract table name from warehouse path pattern /warehouse/db.name/ or similar
+            if (path.contains(".db/")) {
+              int dbIndex = path.lastIndexOf(".db/");
+              String tablePart = path.substring(dbIndex + 4);
+              // Remove trailing slashes
+              tablePart = tablePart.replaceAll("/+$", "");
+              // Construct the full table name including db
+              int warehouseIndex = path.lastIndexOf("/warehouse/");
+              if (warehouseIndex >= 0) {
+                String dbPart = path.substring(warehouseIndex + 11, dbIndex);
+                tableName = dbPart + "." + tablePart;
+                log.info("Extracted table name from warehouse path: {}", tableName);
+                break;
+              }
+            }
+          }
+        }
+      }
+    }
+    
+    // Prepare job names - one for display and one for the URN
+    String jobNameForDisplay = job.getName();
+    String jobNameForUrn = job.getName();
+    
+    // If this is a merge command with an identified table, include the table name
+    if (isMergeIntoCommand && tableName != null && datahubConf.isEnhancedMergeIntoExtraction()) {
+      // Create modified job names that include the table name
+      String tablePart = tableName.replace(".", "_").replace(" ", "_").toLowerCase();
+      String enhancedJobName = job.getName() + "." + tablePart;
```

Comment:
Does jobname contain the operation (I wonder if we should add `merge_into` to the job name to identify these jobs